spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/channel/PostgresChannelMessageTableSubscription.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.channel;
+
+import org.springframework.integration.jdbc.store.JdbcChannelMessageStore;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+
+public final class PostgresChannelMessageTableSubscription {
+
+	private final Object groupId;
+
+	private final String region;
+
+	private final Runnable onPossibleUpdate;
+
+	public PostgresChannelMessageTableSubscription(Object groupId, Runnable onPossibleUpdate) {
+		this(groupId, JdbcChannelMessageStore.DEFAULT_REGION, onPossibleUpdate);
+	}
+
+	public PostgresChannelMessageTableSubscription(Object groupId, String region, Runnable onPossibleUpdate) {
+		this.groupId = groupId;
+		this.region = region;
+		this.onPossibleUpdate = onPossibleUpdate;
+	}
+
+	public Object getGroupId() {
+		return this.groupId;
+	}
+
+	public String getRegion() {
+		return this.region;
+	}
+
+	public void onPossibleUpdate() {
+		this.onPossibleUpdate.run();
+	}
+
+	public static PostgresChannelMessageTableSubscription ofMessageStore(
```

Comment:
I'm still against this solution.
I'd like to see a subscription interface and some `SubscribableChannel` implementation.

Reply:
I redid it to yield a `SubscribableChannel`. Of course, the implementation could be changed to only allow for this abstraction, but this would take away the possibility to being informed that there are new messages without polling them off the queue.